flink-metrics/flink-metrics-jmx/src/main/java/org/apache/flink/metrics/jmx/JMXReporter.java

```diff
@@ -518,9 +525,13 @@ private void startJmxService(int port) throws IOException {
 				throw new IllegalArgumentException("Malformed service url created " + serviceUrl, e);
 			}
 
-			connector = JMXConnectorServerFactory.newJMXConnectorServer(url, null, ManagementFactory.getPlatformMBeanServer());
+			rmiServer = new RMIJRMPServerImpl(port, null, null, null);
+
+		        // Create the connector server now.
+			connector = new RMIConnectorServer(url, null, rmiServer, ManagementFactory.getPlatformMBeanServer());
```

Comment:
Similar as the above comment: Does this one here need to be explicitly closed in the `stop()` method?

Reply:
The connector is already stopped:

if (connector != null) {
				try {
					connector.stop();
				} finally {
					connector = null;
				}
			}